python/src/functions/generate_treasury_report.py

```diff
@@ -229,31 +229,39 @@ def process_event(payload: ProjectLambdaPayload, logger: structlog.stdlib.BoundL
                 file=new_output_file,
             )
         # Output CSV file for treasury
-        with tempfile.NamedTemporaryFile("r+") as csv_file:
+        with tempfile.NamedTemporaryFile(
+            mode="r+t", encoding="utf-8", delete_on_close=False
```

Comment:
Does this need to be `r+t`? Can it be just `w` at this point as it gets re-opened later for reading.

Reply:
yup since there is no reading required in this stage this makes sense. I updated to `wt` 👍